### PR TITLE
UsesGitHubForDevelopment data provider seems to provide false-positives

### LINKED ISSUE #194: UsesGitHubForDevelopment data provider seems to provide many false-positives
The feature 'If a project uses GitHub as the main development platform' is almost always No:

https://github.com/SAP/fosstars-rating-core/blob/master/src/main/jupyter/oss/security/SecurityRatingAnalysis.ipynb

But many of the projects use GitHub for development. This looks like a bug in the data provider.

### PR DESCRIPTION
**Pull request by sourabhsparkala**
- Re-modified checks and removed unwanted ones

This fixes #194

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java [hunk 1]
```diff
@@ -29,31 +29,19 @@
...
   private static final List<Predicate<GHRepository>> CHECKS = Arrays.asList(
 
-      // check if the repository doesn't have an explicit link to the original repository
-      repository -> StringUtils.isEmpty(repository.getMirrorUrl()),
-
       // check if the description mentions "mirror"
       repository -> !repository.getDescription().toLowerCase().contains("mirror"),
 
-      // if GitHub issues are enabled, then it's likely that the project uses GitHub
-      GHRepository::hasIssues,
```
Comment (artem-smotrakov) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java:39]: Why do you drop this check?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java [hunk 2]
```diff
@@ -101,13 +89,29 @@ protected Value fetchValueFor(GitHubProject project) throws IOException {
...
+
+  /**
+   * Check if the URL is not a GitHub URL.
+   * 
+   * @param url Input string.
+   * @return true if the URL is not GitHub URL. Otherwise false.
+   */
+  private static boolean notGitHubUrl(String url) {
+    return !StringUtils.isEmpty(url) 
+        && !url.contains("github.com");
```
[the remark this answers is not in the document]
Comment (sourabhsparkala) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java:115]: Yes that makes sense.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java (hunk 1 above)
Comment (sourabhsparkala) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java:39]: Well with `hasIssues()`, the output gave more False Positives
Like all `github/eclipse` projects were giving output as `true`. Which is a False Positive. Like https://git.eclipse.org/c/?q=jetty.project.

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java:39]: I think Jetty Project does use GitHub for development:

- https://www.eclipse.org/jetty/documentation/current/contributing-patches.html mentions only GitHub
- They take advantage of many GitHub feature such as issue templates https://github.com/eclipse/jetty.project/tree/jetty-10.0.x/.github/ISSUE_TEMPLATE

Comment (sourabhsparkala) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java:39]: You are right about this https://github.com/eclipse/jetty.project/blob/jetty-10.0.x/CONTRIBUTING.md#developer-resources confirms that the jetty.project uses GitHub.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sourabhsparkala) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java]: But the `URL()` does not parse git://git.eclipse.org/gitroot/www.eclipse.org/eclipse.git. Hence, using the `contains` check.

**Comment by sourabhsparkala** [comment]
Tested with the below repositories
-------------------------------------------------------------------------------------------------------------------

All are True Positive

List size : 37
https://github.com/FasterXML/jackson-databind Uses GitHub for development : true
https://github.com/FasterXML/jackson Uses GitHub for development : true
https://github.com/bcgit/bc-java Uses GitHub for development : false
https://github.com/apache/systemml Uses GitHub for development : false
https://github.com/apache/geode Uses GitHub for development : false
https://github.com/apache/hadoop Uses GitHub for development : false
https://github.com/apache/isis Uses GitHub for development : false
https://github.com/apache/karaf Uses GitHub for development : false
https://github.com/apache/camel Uses GitHub for development : false
https://github.com/apache/hive Uses GitHub for development : false
https://github.com/apache/nifi-minifi-cpp Uses GitHub for development : false
https://github.com/apache/activemq Uses GitHub for development : false
https://github.com/apache/poi Uses GitHub for development : false
https://github.com/apache/nifi Uses GitHub for development : false
https://github.com/spring-projects/spring-framework Uses GitHub for development : true
https://github.com/spring-projects/spring-boot Uses GitHub for development : true
https://github.com/spring-projects/spring-security Uses GitHub for development : true
https://github.com/spring-projects/spring-amqp Uses GitHub for development : true
https://github.com/spring-projects/spring-integration Uses GitHub for development : true
https://github.com/spring-projects/spring-vault Uses GitHub for development : true
https://github.com/spring-projects/spring-kafka Uses GitHub for development : true
https://github.com/spring-projects/spring-data-jpa Uses GitHub for development : true
https://github.com/spring-projects/spring-security-oauth Uses GitHub for development : true
https://github.com/spring-projects/spring-data-rest Uses GitHub for development : true
https://github.com/eclipse/mwe Uses GitHub for development : true
https://github.com/eclipse/californium Uses GitHub for development : true
https://github.com/eclipse/jetty.project Uses GitHub for development : true
https://github.com/eclipse/sumo Uses GitHub for development : true
https://github.com/eclipse/mosquitto Uses GitHub for development : true
https://github.com/eclipse/che Uses GitHub for development : true
https://github.com/eclipse/omr Uses GitHub for development : true
https://github.com/eclipse/rdf4j Uses GitHub for development : true
https://github.com/eclipse/jetty.toolchain Uses GitHub for development : true
https://github.com/eclipse/jgit Uses GitHub for development : false
https://github.com/eclipse/egit-github Uses GitHub for development : false
https://github.com/eclipse/eclipse Uses GitHub for development : false
https://github.com/eclipse/egit Uses GitHub for development : false

**Comment by sourabhsparkala** [comment]
Alternately, considered below mentioned URLs
---------------------------------------------------------------------------------------------------------------

- [Open source projects with mirrors on GitHub](https://help.github.com/en/github/getting-started-with-github/finding-ways-to-contribute-to-open-source-on-github#open-source-projects-with-mirrors-on-github)
  - It gives a list of OSS projects which might have mirror projects. E.g: [Eclipse Foundation](https://github.com/eclipse)
  - It also mentions two techniques
    - `mirror:true` search -> A mirror flag indicating if the project is a mirror of the original. Which is already implemented.
    - Post-receive hook -> Gather information from project webhooks. But this is not possible. In order to get the hooks information, one needs to own the project. After some discussion with a GitHub support expert. It was clarified that it would be impossible to get the information if the code push is triggered by a hook or if it is a manual task.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java (hunk 1 above)
Comment (sourabhsparkala) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java:39]: This check is re-added.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java [hunk 3]
```diff
@@ -101,13 +92,28 @@ protected Value fetchValueFor(GitHubProject project) throws IOException {
...
-}
+
+  /**
+   * Check if the URL is not a GitHub URL.
+   * 
+   * @param url Input string.
+   * @return true if the URL is not GitHub URL. Otherwise false.
+   */
+  private static boolean notGitHubUrl(String input) {
+    return !StringUtils.isEmpty(input) && !input.contains("github.com");
```
Comment (artem-smotrakov) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java:117]: Could you please use `URL.getHost()`?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java]: Please then make sure that it check that the host is `github.com`. The following URL passes the check but it's not a URL on GitHub:

https://something.else/github.com/path

[the remark this answers is not in the document]
Comment (sourabhsparkala) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java]: Updated this now.

**Pull request merged**